Sources/MoyaSugar/MoyaSugarProvider.swift

```diff
@@ -19,9 +22,8 @@ open class MoyaSugarProvider<Target: SugarTargetType>: MoyaProvider<Target> {
         url: target.url.absoluteString,
         sampleResponseClosure: endpoint.sampleResponseClosure,
         method: endpoint.method,
-        parameters: endpoint.parameters,
-        parameterEncoding: endpoint.parameterEncoding,
-        httpHeaderFields: endpoint.httpHeaderFields ?? target.httpHeaderFields 
+        task: endpoint.task,
+        httpHeaderFields: endpoint.httpHeaderFields ?? target.headers
```

Comment:
From Moya 9, `MoyaProvider.defaultEndpointMapping` does this job by default. We don't need this code anymore. Just `endpoint.httpHeaderFields` seems to be enough.